packages/attune/src/server/repo/dist/edit.rs

```diff
@@ -0,0 +1,156 @@
+
+use super::{decode_dist_name, super::decode_repo_name};
+
+/// Request to update metadata for an existing distribution.
+///
+/// Only the provided fields will be updated - omitted fields remain unchanged.
+/// This allows partial updates without requiring all metadata to be resent.
+/// Changes take effect immediately but may not be visible in repository indexes
+/// until the next index generation.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct EditDistributionRequest {
+    /// Human-readable description of this distribution.
+    /// APT example: "Debian 11 (bullseye) - Stable Release"
+    pub description: Option<String>,
+    
+    /// The organization or entity that produces this distribution.
+    /// This appears in package manager output and helps users identify the source.
+    /// Examples: "Debian", "Ubuntu", "ACME Corp"
+    pub origin: Option<String>,
+    
+    /// A label for categorizing the distribution.
+    /// Often the same as origin, but can differ for sub-projects or specialized channels.
+    /// Examples: "Debian", "Debian-Security", "Ubuntu"
+    pub label: Option<String>,
+    
+    /// The version number of this distribution release.
+    /// APT examples: "11.0" for Debian 11, "22.04" for Ubuntu 22.04 LTS
+    pub version: Option<String>,
+    
+    /// The suite name indicates the stability level or release channel.
+    /// Common patterns include stability tiers (stable, testing, unstable) or
+    /// update channels (release, updates, security).
+    /// APT examples: "stable", "testing", "unstable", "oldstable", "experimental"
+    pub suite: Option<String>,
+    
+    /// The codename is a unique identifier for a specific release version.
+    /// This provides version stability - tools can reference a specific release
+    /// regardless of its current stability status.
+    /// APT examples: Debian uses "bullseye", "bookworm"; Ubuntu uses "focal", "jammy"
+    pub codename: Option<String>,
+}
+
+/// Response after successfully updating a distribution's metadata.
+///
+/// Returns the distribution ID and name for confirmation. The updated metadata
+/// is immediately available through the API, though changes may not be reflected
+/// in the repository indexes until the next index generation cycle.
+#[derive(Serialize, Deserialize, Debug, Builder)]
+pub struct EditDistributionResponse {
+    /// Unique database identifier for this distribution.
+    pub id: i64,
+    
+    /// The distribution name, confirming which distribution was updated.
+    #[builder(into)]
+    pub distribution: String,
+}
+
+#[axum::debug_handler]
+#[instrument(skip(state))]
+pub async fn handler(
+    State(state): State<ServerState>,
+    tenant_id: TenantID,
+    Path((repository_name, distribution_name)): Path<(String, String)>,
+    Json(req): Json<EditDistributionRequest>,
+) -> Result<Json<EditDistributionResponse>, ErrorResponse> {
+    let repository_name = decode_repo_name(&repository_name)?;
```

Comment:
I wonder if we should make the repository query part of `decode_repo_name`. I don't think we have any instances where we decode the repository name but don't also fetch the repository.